internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -396,6 +405,19 @@ func resourceLinuxVirtualMachine() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if features.FourPointOhBeta() {
```

Comment:
is there a reason why this is limited to 4.0?

Reply:
Hey @catriona-m, thank you for your review.
`disk_controller_type` is `optional+computed`, which may cause a diff after upgrading the provider for users in current provider version. I have added a comment to explain it.

Reply:
it doesn't appear to be optional and computed at the moment?

Reply:
But the value is not optional computed here & if we know the default why can we not simply just configure that be default?

i still don't understand why this needs to wait till 4.0.

Reply:
+1 this should be defaulted on our side - there's no need for this to be computed/wait until 4.0

Reply:
Hi @katbyte, @tombuildsstuff, 
If `disk_controller_type` is not set, its value in the `Get-Response` depends on the value of `size`, for example:

- if `size = Standard_E2bds_v5`, `disk_controller_type` will be returned as `SCSI`
- if `size = Standard_E112ibds_v5`, `disk_controller_type` will be returned as `NVMe`

Since `disk_controller_type` is not fixed for different `size` values, I cannot set a `default` value for it.

Reply:
> you could just make it computed thou? while we would rather have a proper default in this case as we can't computed should be fine? why would making it computed cause a diff for existing users? it would just adopt the value they have picked?

Do you mean I can define optional+computed for this property? I thought I cannot use that way because this [best practice](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed) does not allow optional+computed anymore.

Reply:
while generally yes that is correct and we shouldn't, there still are times where it has to be done because theres no "correct" way to dynamically specify a default. so in cases like this were the only way to get "the default" is from the API we have to use computed still.

Reply:
code updated. Thanks.